includes/options/class-amp-options-menu.php

```diff
@@ -573,4 +574,47 @@ public function render_screen() {
 		</div>
 		<?php
 	}
+
+	/**
+	 * Possibly replace the message of the 'Settings saved' notice.
+	 *
+	 * If this is the first time a user has changed a setting here,
+	 * display 'View your site as AMP...' instead of 'Settings saved'.
+	 * Altering a setting in the global $wp_settings_errors is not ideal.
+	 * But it looks like there is no filter for this.
+	 */
+	public function possibly_replace_settings_saved_notice() {
+		global $wp_settings_errors;
+		$meta_key = 'amp_view_your_site_notice';
+		$user_id  = get_current_user_id();
+
+		// Exit if this setting isn't present, or if this notice has already displayed.
+		if ( ! get_settings_errors( 'general' ) || get_user_meta( $user_id, $meta_key, true ) ) {
+			return;
+		}
+
+		foreach ( $wp_settings_errors as $key => $setting ) {
+			if ( 'general' === $setting['setting'] && 'settings_updated' === $setting['code'] ) {
```

Comment:
If you call `add_settings_error()` inside the option sanitizer callback, where we can detect the mode change, then this will get stored in a transient to then automatically be displayed with the `admin_notices` when the page reloads.

What this won't do is hide the default settings updated notice.

Note that `add_settings_error()` has a `$type` parameter which can be `"updated"`, so not an error at all.

Reply:
Great idea. https://github.com/Automattic/amp-wp/pull/1443/commits/cc2e62fbfcbc21d980a509dbe4276bc9e95f7c58 uses `add_settings_error()` in the `sanitize_callback`. 

Though I'm not sure why, it seems like the default message doesn't display when this is added. 

<img width="1013" alt="on-paired-mode" src="https://user-images.githubusercontent.com/4063887/45719762-f45c6b80-bb77-11e8-9e69-e71c99c5b6ee.png">